### PR TITLE
Default to use netty bytebuf in network layer and remove getAndRelease method

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This pr includes two changes:

1. Default to use netty bytebuf in network layer, which will remove lots of code dealing with java bytebuffer
2. Remove getAndRelease and replace it with content method and implements ByteBufHolder interface.

### CODE REVIEW

File: ambry-network/src/main/java/com.github.ambry.network/LocalRequestResponseChannel.java [hunk 1]
```diff
@@ -121,14 +122,13 @@ public void shutdown() {
    * Utility to extract a byte buffer from a {@link Send} object, skipping the size header.
    * @param payload the payload whose bytes we want.
    */
-  static ByteBuffer byteBufferFromPayload(Send payload) throws IOException {
+  static ByteBuf byteBufFromPayload(Send payload) throws IOException {
```
Comment (jsjtzyy) [inline ambry-network/src/main/java/com.github.ambry.network/LocalRequestResponseChannel.java]: minor: update java doc

File: ambry-network/src/main/java/com.github.ambry.network/SocketRequestResponseChannel.java [hunk 2]
```diff
@@ -53,21 +54,24 @@ public long getStartTimeInMs() {
...
+
+  @Override
+  public ByteBuf content() {
+    return content;
+  }
+
+  @Override
+  public SocketServerRequest replace(ByteBuf content) {
+    SocketServerRequest request = new SocketServerRequest(getProcessor(), getConnectionId(), content);
+    return request;
```
Comment (jsjtzyy) [inline ambry-network/src/main/java/com.github.ambry.network/SocketRequestResponseChannel.java]: same here

File: ambry-network/src/main/java/com.github.ambry.network/SocketServer.java [hunk 3]
```diff
@@ -419,9 +420,8 @@ public void run() {
         List<NetworkReceive> completedReceives = selector.completedReceives();
         for (NetworkReceive networkReceive : completedReceives) {
           String connectionId = networkReceive.getConnectionId();
-          Object buffer = networkReceive.getReceivedBytes().getAndRelease();
-          SocketServerRequest req = new SocketServerRequest(id, connectionId, buffer,
-              Utils.createDataInputStreamFromBuffer(buffer, networkConfig.networkPutRequestShareMemory));
+          ByteBuf buffer = networkReceive.getReceivedBytes().content();
+          SocketServerRequest req = new SocketServerRequest(id, connectionId, buffer);
```
Comment (jsjtzyy) [inline ambry-network/src/main/java/com.github.ambry.network/SocketServer.java]: who is supposed to release the buffer?

File: ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java [hunk 4]
```diff
@@ -212,11 +212,10 @@ public void testNormalOperation() throws Exception {
 
       // handle any responses we may have gotten
       for (NetworkReceive receive : selector.completedReceives()) {
-        ByteBuffer payload = (ByteBuffer) (receive.getReceivedBytes().getAndRelease());
+        ByteBuf payload = receive.getReceivedBytes().content();
```
Comment (jsjtzyy) [inline ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java]: did we release the payload?

File: ambry-network/src/main/java/com.github.ambry.network/SocketServer.java (hunk 3 above)
Comment (justinlin-linkedin) [reply ambry-network/src/main/java/com.github.ambry.network/SocketServer.java]: in the RequestHandler, we have a for loop to get the request out and after dealing with the request, we release it.

File: ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java (hunk 4 above)
Comment (justinlin-linkedin) [reply ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java]: good catch. updated

**Pull request merged**